lib/private/Group/Database.php

```diff
@@ -377,6 +381,63 @@ public function usersInGroup($gid, $search = '', $limit = -1, $offset = 0) {
 		return $users;
 	}
 
+	public function searchInGroup(string $gid, string $search = '', int $limit = -1, int $offset = 0): array {
+		$this->fixDI();
+
+		$query = $this->dbConn->getQueryBuilder();
+		if ($search !== '') {
+			$query->select('g.uid', 'u.displayname');
+		} else {
+			$query->select('g.uid');
+		}
+
+		$query->from('group_user', 'g')
+			->where($query->expr()->eq('gid', $query->createNamedParameter($gid)))
+			->orderBy('g.uid', 'ASC');
+
+		if ($search !== '') {
+			$query->leftJoin('g', 'users', 'u', $query->expr()->eq('g.uid', 'u.uid'))
+				->leftJoin('u', 'preferences', 'p', $query->expr()->andX(
+					$query->expr()->eq('p.userid', 'u.uid'),
+					$query->expr()->eq('p.appid', $query->expr()->literal('settings')),
+					$query->expr()->eq('p.configkey', $query->expr()->literal('email')))
+				)
+				// sqlite doesn't like re-using a single named parameter here
+				->andWhere(
+					$query->expr()->orX(
+						$query->expr()->ilike('g.uid', $query->createNamedParameter('%' . $this->dbConn->escapeLikeParameter($search) . '%')),
+						$query->expr()->ilike('u.displayname', $query->createNamedParameter('%' . $this->dbConn->escapeLikeParameter($search) . '%')),
+						$query->expr()->ilike('p.configvalue', $query->createNamedParameter('%' . $this->dbConn->escapeLikeParameter($search) . '%'))
+					)
+				)
+				->orderBy('u.uid_lower', 'ASC');
+		}
+
+		if ($limit !== -1) {
+			$query->setMaxResults($limit);
+		}
+		if ($offset !== 0) {
+			$query->setFirstResult($offset);
+		}
+
+		$result = $query->executeQuery();
+
+		$users = [];
+		$userManager = \OC::$server->get(IUserManager::class);
+		$displayNameCache = \OC::$server->get(DisplayNameCache::class);
+		while ($row = $result->fetch()) {
+			if (isset($row['displayname'])) {
+				$users[] = new LazyUser($row['uid'], $displayNameCache, $userManager, $row['displayname']);
+			} else {
+				// TODO maybe also fetch the displayname directly here
+				$users[] = new LazyUser($row['uid'], $displayNameCache, $userManager);
+			}
```

Comment:
```suggestion
			// TODO maybe also fetch the displayname directly here
			$users[] = new LazyUser($row['uid'], $displayNameCache, $userManager, $row['displayname'] ?? null);
```

Also why $displayNameCache and $userManager are not get from \OC::$server in LazyUser class instead?

Reply:
Because then you need to get the value for each iteration of the loop